packages/core/core/src/requests/AssetRequestRunner.js

```diff
@@ -0,0 +1,155 @@
+// @flow strict-local
+import type WorkerFarm from '@parcel/workers';
+import type AssetGraph from '../AssetGraph';
+import type RequestTracker, {RequestRunnerAPI} from '../RequestTracker';
+import type {
+  AssetRequestDesc,
+  AssetRequestResult,
+  ParcelOptions,
+  TransformationOpts,
+} from '../types';
+
+import invariant from 'assert';
+import nullthrows from 'nullthrows';
+import {RequestRunner, generateRequestId} from '../RequestTracker';
+
+export type AssetRequest = {|
+  id: string,
+  +type: 'asset_request',
+  request: AssetRequestDesc,
+  result?: AssetRequestResult,
+|};
+
+export default class AssetRequestRunner extends RequestRunner<
+  AssetRequestDesc,
+  AssetRequestResult,
+> {
+  options: ParcelOptions;
+  optionsRef: number;
+  configRef: number;
+  runTransform: TransformationOpts => Promise<AssetRequestResult>;
+  assetGraph: AssetGraph;
+
+  constructor(opts: {|
+    tracker: RequestTracker,
+    options: ParcelOptions,
+    optionsRef: number,
+    configRef: number,
+    workerFarm: WorkerFarm,
+    assetGraph: AssetGraph,
+  |}) {
+    super(opts);
+    this.type = 'asset_request';
+    this.options = opts.options;
+    this.optionsRef = opts.optionsRef;
+    this.configRef = opts.configRef;
+    this.runTransform = opts.workerFarm.createHandle('runTransform');
+    this.assetGraph = opts.assetGraph;
+  }
+
+  async run(request: AssetRequestDesc, api: RequestRunnerAPI) {
+    api.invalidateOnFileUpdate(
+      await this.options.inputFS.realpath(request.filePath),
+    );
+    let start = Date.now();
+    let {assets, configRequests} = await this.runTransform({
+      request: request,
+      optionsRef: this.optionsRef,
```

Comment:
why do instance methods (this and `runValidate` in `AssetGraphBuilder`) need to be passed instance properties? do the properties change over time?

Reply:
`runTransform` and `runValidate` are both worker handles that don't have access to instance properties.